### PR TITLE
Update IQ-TREE to 1.6.6

### PR DESCRIPTION
**Pull request by apcamargo**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/iqtree/meta.yaml [hunk 1]
```diff
@@ -19,15 +19,18 @@ requirements:
...
     - eigen
     - zlib
+    - openmp # [osx]
   run:
     - zlib
+    - openmp # [osx]
 
 test:
   commands:
-    - iqtree-omp
```
Comment (bgruening) [inline recipes/iqtree/meta.yaml]: Was this dropped from the tool?

**Comment by apcamargo** [comment]
Yes. Since version 1.6.0, the two versions of IQ-TREE  were merged.

http://www.iqtree.org/release/

**Comment by apcamargo** [comment]
I can't get IQ-TREE to build on macOS. I tried to add OpenMP as a dependency but it didn't work.

The compilation guide states: 

> To compile the multicore version, the default installed Clang unfortunately does not support OpenMP (which might change in the near future). However, the latest Clang 3.7 supports OpenMP, which can be downloaded from http://clang.llvm.org.
http://www.iqtree.org/doc/Compilation-Guide

Could the Clang version used in the CI not support OpenMP?

File: recipes/iqtree/meta.yaml (hunk 1 above)
Comment (apcamargo) [reply recipes/iqtree/meta.yaml]: Yes. Since version 1.6.0, the two versions of IQ-TREE were merged.

http://www.iqtree.org/release/

**Comment by tseemann** [comment]
@apcamargo i did the brew formula for iqtree, i had to patch it even for gcc!
```
    if OS.mac?
      inreplace "CMakeLists.txt",
        "${CMAKE_EXE_LINKER_FLAGS_RELEASE} -Wl,--gc-sections",
        "${CMAKE_EXE_LINKER_FLAGS_RELEASE}"
    end
```

**Comment by apcamargo** [comment]
Hi @tseemann . I'm sorry, I don't think I understood your suggestion. As far as I know, the cmake command you suggested is eqivalent to the one in the build file. Am I missing something?

**Comment by tseemann** [comment]
As openmp isn't supported on MacOS, you might need to remove the `IQTREE_FLAGS=omp` to build on Mac.  Or make it linux only.

**Comment by bgruening** [comment]
@apcamargo LGTM! ping @bioconda/osx for help regarding the OSX if needed.

**Comment by bgruening** [comment]
We could use GCC on OSX as well if @bioconda/osx does not have a better idea.

**Comment by tseemann** [comment]
@bgruening how do you force OSX to use gcc instead of clang for C++ code?
 `compiler("gcc")` ?

**Comment by bgruening** [comment]
@tseemann you can overwrite the globals by putting a https://github.com/bioconda/bioconda-recipes/blob/master/recipes/mawk/conda_build_config.yaml next to your meta.yaml file and replace clang with gcc.

**Comment by apcamargo** [comment]
@tseemann @bgruening I've submitted [another PR](https://github.com/bioconda/bioconda-recipes/pull/10583). I think we can continue our discussion there.

**Pull request merged**